homeassistant/components/tuya/__init__.py

```diff
@@ -50,17 +53,35 @@
 )
 
 
-def setup(hass, config):
+def setup(hass, config, retry_delay=FIRST_RETRY_TIME):
     """Set up Tuya Component."""
 
+    _LOGGER.debug("Setting up integration")
+
     tuya = TuyaApi()
     username = config[DOMAIN][CONF_USERNAME]
     password = config[DOMAIN][CONF_PASSWORD]
     country_code = config[DOMAIN][CONF_COUNTRYCODE]
     platform = config[DOMAIN][CONF_PLATFORM]
 
+    try:
+        tuya.init(username, password, country_code, platform)
+    except (TuyaNetException, TuyaServerException):
```

Comment:
What happens if the credentials are bad?

Reply:
It generate different exception and fail

Reply:
We should catch that and `return False`.

Reply:
Done. Now all check are OK.